ctags.go

```diff
@@ -257,6 +261,11 @@ func (p *ctagsProcess) Parse(name string, content []byte) ([]*Entry, error) {
 				Pattern:    rep.Pattern,
 				Signature:  rep.Signature,
 			})
+		case "completed":
+			if len(symbolErrors) > 0 {
+				return es, &ParseError{Message: fmt.Sprintf("failed to parse some symbols:\n\t%s", strings.Join(symbolErrors, "\n\t"))}
```

Comment:
It's a bit strange from an API perspective to return a non-nil error and a value to consume. I think we need to rethink the API here. Maybe a response of `(*Result, error)` where Result is
```go
struct {
  Entries []*Entry
  Warnings []string
}
```
or alternatively we use the old method of just logging them? Or maybe have some other way to report warnings?

What did you want to do with these warnings again?

Reply:
Sounds good to me! I agree this json protocol is not great and we should explore something much better.